measurement/plugins/latency/measurements.py

```diff
@@ -0,0 +1,232 @@
+import re
+import subprocess
+
+import validators
+from validators import ValidationFailure
+
+from measurement.measurements import BaseMeasurement
+from measurement.plugins.latency.results import (
+    LatencyMeasurementResult,
+    LatencyIndividualMeasurementResult,
+)
+from measurement.results import Error
+from measurement.units import RatioUnit, TimeUnit, StorageUnit
+
+LATENCY_OUTPUT_REGEX = re.compile(
+    r"= (?P<minimum_latency>[\d.].*)/(?P<average_latency>[\d.].*)/(?P<maximum_latency>[\d.].*)/(?P<median_deviation>[\d.].*) "
+)
+LATENCY_PACKETS_REGEX = re.compile(
+    r"(?P<packets_transmitted>\d*) packets transmitted, (?P<packets_received>\d*) received, "
+    r"(?P<packet_loss>\d*)% packet loss, time (?P<time>\d*)(?P<time_unit>.*)"
+)
+LATENCY_INDIVIDUAL_PING_REGEX = re.compile(
+    r"(?P<packet_size>\d*) (?P<packet_size_units>.*) from (?P<reverse_dns_address>.*) \(("
+    r"?P<ip_address>.*)\): icmp_seq=(?P<icmp_sequence>\d*) ttl=(?P<time_to_live>\d*) time=("
+    r"?P<time>.*) (?P<time_unit>.*)"
+)
+
+
+LATENCY_ERRORS = {
+    "ping-err": "ping had an unknown error",
+    "ping-split": "ping attempted to split the result but it was in an unanticipated format",
+    "ping-regex": "ping attempted get the known regex format and failed.",
+    "ping-minimum-latency": "ping could not process the minimum latency.",
+    "ping-maximum-latency": "ping could not process the maximum latency.",
+    "ping-average-latency": "ping could not process the average latency.",
+    "ping-median-deviation": "ping could not process the median deviation.",
+    "ping-no-server": "No closest server could be resolved.",
+    "ping-timeout": "Measurement request timed out.",
+}
+
+
+class LatencyMeasurement(BaseMeasurement):
+    def __init__(self, id, host, count=4, include_individual_results=False):
+        super(LatencyMeasurement, self).__init__(id=id)
+        if count < 1:
+            raise ValueError(
+                "A value of {count} was provided for the number of pings. This must be a positive "
+                "integer greater than 0.".format(count=count)
+            )
+
+        validated_domain = validators.domain(host)
+        if isinstance(validated_domain, ValidationFailure):
+            raise ValueError("`{host}` is not a valid host".format(host=host))
+
+        self.host = host
+        self.count = count
+        self.include_individual_results = include_individual_results
+
+    def measure(self):
+        return self._get_latency_results(
+            self.host,
+            count=self.count,
+            include_individual_results=self.include_individual_results,
+        )
+
+    def _get_latency_results(self, host, count=4, include_individual_results=False):
+        """Perform the latency measurement."""
```

Comment:
It would be helpful this to outline what this returns - it changed from a single result to a list which was a bit confusing.